extensions/TheShovel/ShovelUtils.js

```diff
@@ -328,8 +351,25 @@
     getfps(){
       return fps;
     }
-  }
 
+    deleteImage({ SPRITE, COSNAME }){
+      //0znzw, since shovel did not add it yet.
+      const target = vm.runtime.getSpriteTargetByName(SPRITE);
+      if (!target) {
+        return;
+      }
+      target.deleteCostume(target.getCostumeIndexByName(COSNAME));
```

Comment:
did you test what happens if no costume with the name exists?

have you tested what happens if you try to delete every costume in the sprite?

Reply:
yes for both

Reply:
> have you tested what happens if you try to delete every costume in the sprite?

it keeps 1 as scratch does not let you have less than 1.

Reply:
> did you test what happens if no costume with the name exists?

nothing happens

Reply:
i did not mean to mark this one as resolved,

Reply:
neither cases produce errors, and have been tested